daemon/main.go

```diff
@@ -693,6 +693,10 @@ func initEnv(cmd *cobra.Command) {
 	}
 
 	log.Infof("Container runtime options set: %s", workloads.GetRuntimeOptions())
+
+	if viper.GetBool("sidecar-http-proxy") {
```

Comment:
You can use `flags.MarkHidden("sidecar-http-proxy")` to hide it from usage but still support setting the flag.

Reply:
That's always been the case for this flag.